CMakeLists.txt

```diff
@@ -61,6 +57,15 @@ endif()
 list (APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 list (APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
 
+# parse the current version based on git tags
+include(GetGitRevisionDescription)
+git_describe(GIT_VERSION --tags)
+string(REGEX REPLACE "-([0-9]*).*" ".\\1" VERSION "${GIT_VERSION}")
+
+# configure project name and version
+project (xiphos
+	VERSION "${VERSION}")
```

Comment:
If this isn't working then maybe try the set_property command?

Reply:
I guess it depends of what you mean by "working". It's setting the version properly: the built software gets the right version, the logs show show the right version, and the packaged filenames have the right version. The problem is that it then bundles a copy of itself inside the source package, and _THAT_ can't figure anything out because `VERSION` isn't actually set anywhere it is derived from git repository meta data — which of course doesn't exist in the tarball.

From my experience with GNU-Make the way to fix this is probably to cache the version in a file somewhere during normal builds and use the cache to retrieve it in the event `.git`, I just don't know how to do that in Cmake.

Reply:
Ah, yes. It would be pretty normal to set the value statically when you're cutting a release. I would suggest writing a file into the cmake/ directory. Then you can check for its existence and use that if found. If you don't find it, then use the git method as a fallback.

Reply:
I think I got it. Finally.